### ISSUE TITLE
Make the DOM elements of the selection easily available to extensions

### ISSUE THREAD

**Issue by wjt** [openannotation/annotator#549]
In my application, I need to store some extra metadata about the selected ranges. (Specifically: I have `data-text-id` attributes for each line of text, so I want to store the start and end ID of the selected text.) I currently do something like this:

```javascript
        beforeAnnotationCreated: function(annotation) {
          annotation['textids'] = annotation.ranges.map(function(range) {
            var domRange = Range.sniff(range).normalize(root);
            if (domRange.limit(options.element) !== null) {
              return {
                start: getTextId(domRange.start),
                end: getTextId(domRange.end)
              }
            } else {
              // fallback path elided
              return null;
            }
          }).filter(function(o) { return !!o; });
        }
```

It's a bit sad that I have to reconstruct information that Annotator has but does not expose. Perhaps `beforeAnnotationCreated` and friends could receive a second dictionary of information like this, derived from the data which will actually be stored? I'm not quite sure what it would look like.

**Comment by tilgovi** [comment]
You're asking for the unserialized range, right?

**Comment by wjt** [comment]
Exactly. In my case, I only really need the endpoints of the range, but the range itself is ideal.

(It would feel a bit strange to pass it in the annotation dictionary, only to be magically stripped out by the storage backend, but that would work of course…)

**Comment by tilgovi** [comment]
I opened #583, #584 and #585 to decompose the lifecycle and introduce a component that you can use to access the highlights directly, which hopefully will give you what you need.
